### PR TITLE
doc: update README to align with the new brand name

### LINKED ISSUE #7: Update the README for the repository
I propose to update the README.md file for the repository with the following information:
- link to the overview of the organization
- purpose of the repository
- is it in active development or not, is it archived or not, any reasons for that
- links to relevant documentation (security audits, web site, design documents, etc.)

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Resolve #7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,17 @@
-# Notation Github Actions
-Github Actions for [Notation](https://notaryproject.dev/).
+# GitHub Actions for Notation
 
-Supported actions: `Notation: Setup`, `Notation: Sign` and `Notation: Verify`.
+This repository contains the implementation of [GitHub Actions](https://docs.github.com/en/actions) for [Notation](https://notaryproject.dev/). It provides multiple actions for signing and verifying OCI artifacts with Notation in CI/CD. This project is still in early development status.
```
Comment (yizha1) [inline README.md:3]: Suggest using the `notation` repo link

**Review by yizha1 — COMMENTED** [review]
Thanks @FeynmanZhou , LGTM, but IANAM

File: README.md (hunk 1 above)
Comment (Two-Hearts) [inline README.md:3]: nit:
```suggestion
This repository contains the implementation of [GitHub Actions](https://docs.github.com/en/actions) for [Notation](https://notaryproject.dev/). It provides actions for signing and verifying OCI artifacts with Notation in CI/CD. This project is still in early development status.
```

**Review by duffney — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -79,6 +87,7 @@ Example of using the [Referrers API](https://github.com/opencontainers/distribut
     trust_store: <dir_to_user_trust_store>
     allow_referrers_api: <boolean_flag_for_referrers_api>
 ```
+
```
Comment (Two-Hearts) [inline README.md:90]: A super nit:
Other `For example` does not have an extra line above. Shall we add them as well?

Comment (FeynmanZhou) [reply README.md]: I am just aware of this extra line. Removed this line.

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Pull request merged**